api/src/wfl/service/terra.clj

```diff
@@ -39,3 +41,11 @@
         response (http/get submission-url {:headers (once/get-auth-header)})]
     (util/parse-json (:body response))))
 
+(defn get-workflow-status-by-entity
+  "Get workflow status given a Terra submission-id and entity-name."
+  [terra-url workspace {:keys [uuid inputs] :as _item}]
+  (->> (get-submission terra-url workspace uuid)
+       :workflows
+       (filter #(= (:entity-name inputs) (get-in % [:workflowEntity :entityName])))
+       (first)
+       (:status)))
```

Comment:
What happens if the submission is not found? does get-submission throw or will this return null?

Reply:
In that case `get-submission` would return a 404 error

Reply:
Thanks!